OP-1193 | Update  ward Inventory

JantBogard opened the pull request:
See [OP-1193](https://openhospital.atlassian.net/browse/OP-1193)

[OP-1193]: https://openhospital.atlassian.net/browse/OP-1193?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

mwithi committed:
Merge pull request #1787 from informatici/release_1_13_0

Release 1.13.0

mwithi committed:
Remove -SNAPSHOT

mwithi committed:
Merge remote-tracking branch 'origin/develop' into release_1_14_0

mwithi committed:
Merge pull request #1967 from informatici/release_1_14_0

Merge develop for release 1.14.0

mwithi committed:
Update version

mwithi committed:
Update README.md

mwithi committed:
Update README.md

mwithi committed:
Merge pull request #1976 from informatici/release_1_14_1

Release_1.14.1

JantBogard committed:
get inventory

JantBogard committed:
Merge commit 'f6070bd0477b56c552254a05acce3375613dc4f3' into OP-1193

JantBogard committed:
Update ward Inventory

dbmalkovsky commented on bundle/language_en.properties:
```diff
@@ -662,6 +664,52 @@ angal.hospital.visitduration.txt
+angal.inventory.nextarrow.btn                                                                          = >
+angal.inventory.noproduct.msg                                                                          = Product is not found.
+angal.inventory.noproductfound.msg                                                                     = No medical found
+angal.inventory.noteditable.msg                                                                            = A validated inventory can not be modified
```
```suggestion
angal.inventory.noteditable.msg                                                                        = A validated inventory cannot be modified.
```

In this diff `can not` is used twice and `cannot` is used 10 times.

Also messages should be statements that end in a period.

dbmalkovsky commented on bundle/language_en.properties:
```diff
@@ -662,6 +664,52 @@ angal.hospital.visitduration.txt
 angal.hospital.visitendhour.txt                                                                        = Visit end hour
 angal.hospital.visitstarthour.txt                                                                      = Visit start hour
 angal.lab.allnegative.txt                                                                              = All negative.
+angal.inventory.acanceledinventorycannotbemodified.msg                                                 = A canceled inventory can not be modified
```
```suggestion
angal.inventory.acanceledinventorycannotbemodified.msg                                                 = A canceled inventory cannot be modified.
```
In this diff `can not` is used twice and `cannot` is used 10 times.

Also messages should be statements that end in a period.

dbmalkovsky commented on bundle/language_en.properties:
```diff
@@ -662,6 +664,52 @@ angal.hospital.visitduration.txt
+angal.inventory.cannotcreateanotherinventorywithotherinprogressinthisward.msg                          = Cannot create another inventory with other in progress in this ward.
+angal.inventory.date.label                                                                             = Inventory's date
+angal.inventory.edit.title                                                                             = Edit an inventory
+angal.inventory.error.msg                                                                              = Error during  some products's recording
```
```suggestion
angal.inventory.error.msg                                                                              = Error during a product's recording.
```

dbmalkovsky commented on bundle/language_en.properties:
```diff
@@ -662,6 +664,52 @@ angal.hospital.visitduration.txt
+angal.inventory.allproduct.radio                                                                       = All products
+angal.inventory.cannotcreateanotherinventorywithotherinprogressinthisward.msg                          = Cannot create another inventory with other in progress in this ward.
+angal.inventory.date.label                                                                             = Inventory's date
+angal.inventory.edit.title                                                                             = Edit an inventory
```
Words in a dialog's title should start with a capital letter.   In this diff there is a mixture of capitalization.

dbmalkovsky commented on bundle/language_en.properties:
```diff
@@ -662,6 +664,52 @@ angal.hospital.visitduration.txt
 angal.lab.allnegative.txt                                                                              = All negative.
+angal.inventory.acanceledinventorycannotbemodified.msg                                                 = A canceled inventory can not be modified
+angal.inventory.allproduct.radio                                                                       = All products
+angal.inventory.cannotcreateanotherinventorywithotherinprogressinthisward.msg                          = Cannot create another inventory with other in progress in this ward.
```
```suggestion
angal.inventory.cannotcreateanotherinventorywithotherinprogressinthisward.msg                          = Cannot create another inventory with another in progress in this ward.
```

dbmalkovsky commented on src/main/java/org/isf/medicalinventory/gui/InventoryWardEdit.java:
```diff
@@ -0,0 +1,1175 @@
+        saveButton = new JButton(MessageBundle.getMessage("angal.common.save"));
+        saveButton.setMnemonic(MessageBundle.getMnemonic("angal.common.save.btn.key"));
+        saveButton.addActionListener(actionEvent -> {
+            String State = InventoryStatus.draft.toString();
```
```suggestion
            String state = InventoryStatus.draft.toString();
```

dbmalkovsky replied:
Why load this into a local variable if the value is only used once and that being inside an `if` statement at line 384?

dbmalkovsky commented on src/main/java/org/isf/medicalinventory/gui/InventoryWardEdit.java:
```diff
@@ -0,0 +1,1175 @@
+                inventory = new MedicalInventory();
+                inventory.setInventoryReference(reference);
+                inventory.setInventoryDate(dateInventory);
+                inventory.setStatus(State);
```
```suggestion
                inventory.setStatus(state);
```

dbmalkovsky commented on src/main/java/org/isf/medicalinventory/gui/MedicalPicker.java:
```diff
@@ -0,0 +1,339 @@
+        this.medicalMapValues = medicals;
+    }
+
+    /**
```
why leave comments like this?  It provides no real value.

dbmalkovsky commented on src/main/java/org/isf/medicalinventory/gui/InventoryWardBrowser.java:
```diff
@@ -0,0 +1,590 @@
+            int selectedRow = jTableInventory.getSelectedRow();
+            if (selectedRow > -1) {
+                inventory = inventoryList.get(selectedRow);
+                if (inventory.getStatus().equals("3")) {
```
Isn't there a `enum` defined for this in the `util/jobjects` package?

JantBogard committed:
Applying suggestion from code review

dependabot[bot] committed:
Chore(deps): Bump org.slf4j:slf4j-reload4j from 2.0.12 to 2.0.13

Bumps org.slf4j:slf4j-reload4j from 2.0.12 to 2.0.13.

---
updated-dependencies:
- dependency-name: org.slf4j:slf4j-reload4j
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Chore(deps): Bump org.apache.maven.plugins:maven-jar-plugin

Bumps [org.apache.maven.plugins:maven-jar-plugin](https://github.com/apache/maven-jar-plugin) from 3.3.0 to 3.4.0.
- [Release notes](https://github.com/apache/maven-jar-plugin/releases)
- [Commits](https://github.com/apache/maven-jar-plugin/compare/maven-jar-plugin-3.3.0...maven-jar-plugin-3.4.0)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-jar-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Chore(deps): Bump org.slf4j:slf4j-api from 2.0.12 to 2.0.13

Bumps org.slf4j:slf4j-api from 2.0.12 to 2.0.13.

---
updated-dependencies:
- dependency-name: org.slf4j:slf4j-api
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

mwithi commented on bundle/language_en.properties:
```diff
@@ -664,19 +664,24 @@ angal.hospital.visitduration.txt
 angal.inventory.date.label                                                                             = Inventory's date
-angal.inventory.edit.title                                                                             = Edit an inventory
-angal.inventory.error.msg                                                                              = Error during  some products's recording
+angal.inventory.edit.title                                                                             = Edit An Inventory
```
```suggestion
angal.inventory.edit.title                                                                             = Edit a Inventory
```

dbmalkovsky replied:
Actually `an` is grammatically correct.     `an` is used when the next work starts with a vowel; otherwise `a` is used.

dbmalkovsky replied:
And why not just `Edit Inventory`?   We don't say `Edit a Patient` do we?

mwithi commented on bundle/language_en.properties:
```diff
@@ -693,11 +698,15 @@ angal.inventory.state.validate.txt
 angal.inventory.unitprice.col                                                                          = Unit price
+angal.inventory.updatesucces.msg                                                                       = Inventory updated.
+angal.inventory.updateerror.msg                                                                        = Inventory not updated.
+angal.inventory.updaterowerror.msg                                                                     = Error during  some products's update.
```
```suggestion
angal.inventory.updaterowerror.msg                                                                     = Error during some products's update.
```

mwithi commented on src/main/java/org/isf/medicalinventory/gui/InventoryWardBrowser.java:
```diff
@@ -84,6 +100,16 @@ public class InventoryWardBrowser extends ModalJFrame implements InventoryListen
     private int[] pColumwidth = { 150, 150, 100, 100, 150 };
     private JComboBox<Object> stateComboBox;
     private JLabel stateLabel;
+    JButton next;
+    JButton previous;
+    JComboBox pagesCombo = new JComboBox();
+    JLabel under = new JLabel("/ 0 Page");
```
Shouldn't these fields be private, pagesCombo be instantiated in a getter, JLabel text been defined in a language bundle...

mwithi commented on src/main/java/org/isf/medicalinventory/gui/InventoryWardBrowser.java:
```diff
@@ -330,6 +471,12 @@ public Class<?> getColumnClass(int c) {
         }
 
+        public int getRowCount() {
+            if (inventoryList == null)
+                return 0;
```
```suggestion
            if (inventoryList == null) {
                return 0;
            }
```

JantBogard committed:
Applying suggestion from review code

dbmalkovsky committed:
OP-1149 Update to Spring Boot 3.2.x (#1979)

* OP-1149 Update to Spring Boot 3.2.x

* OP-1149 Update to Spring Boot 3.2.x

* Update pom.xml update to 1.14.1-SNAPSHOT

dbmalkovsky committed:
Chore: update min java version to JDK 17 (#1977)

dbmalkovsky committed:
Merge pull request #1984 from informatici/dependabot/maven/org.slf4j-slf4j-api-2.0.13

Chore(deps): Bump org.slf4j:slf4j-api from 2.0.12 to 2.0.13

dbmalkovsky committed:
Merge pull request #1982 from informatici/dependabot/maven/org.slf4j-slf4j-reload4j-2.0.13

Chore(deps): Bump org.slf4j:slf4j-reload4j from 2.0.12 to 2.0.13

dependabot[bot] committed:
Chore(deps): Bump org.apache.maven.plugins:maven-compiler-plugin

Bumps [org.apache.maven.plugins:maven-compiler-plugin](https://github.com/apache/maven-compiler-plugin) from 3.12.1 to 3.13.0.
- [Release notes](https://github.com/apache/maven-compiler-plugin/releases)
- [Commits](https://github.com/apache/maven-compiler-plugin/compare/maven-compiler-plugin-3.12.1...maven-compiler-plugin-3.13.0)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-compiler-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dbmalkovsky committed:
Merge pull request #1983 from informatici/dependabot/maven/org.apache.maven.plugins-maven-jar-plugin-3.4.0

Chore(deps): Bump org.apache.maven.plugins:maven-jar-plugin from 3.3.0 to 3.4.0

dbmalkovsky committed:
Merge pull request #1972 from informatici/dependabot/maven/org.apache.maven.plugins-maven-compiler-plugin-3.13.0

Chore(deps): Bump org.apache.maven.plugins:maven-compiler-plugin from 3.12.1 to 3.13.0

dbmalkovsky committed:
OP-1149 The main class is no longer `org.isf.menu.gui.Menu` but `org.isf.Application` (#1985)

dbmalkovsky committed:
OP-1210 quick "fix" for changes from CORE PR-1210. (#1987)

dependabot[bot] committed:
Chore(deps): Bump org.apache.maven.plugins:maven-jar-plugin

Bumps [org.apache.maven.plugins:maven-jar-plugin](https://github.com/apache/maven-jar-plugin) from 3.4.0 to 3.4.1.
- [Release notes](https://github.com/apache/maven-jar-plugin/releases)
- [Commits](https://github.com/apache/maven-jar-plugin/compare/maven-jar-plugin-3.4.0...maven-jar-plugin-3.4.1)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-jar-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dbmalkovsky committed:
Merge pull request #1986 from informatici/dependabot/maven/org.apache.maven.plugins-maven-jar-plugin-3.4.1

Chore(deps): Bump org.apache.maven.plugins:maven-jar-plugin from 3.4.0 to 3.4.1

JantBogard committed:
Applying suggestion from review code

dbmalkovsky committed:
OP-1230: re-enable tests (#1988)

dependabot[bot] committed:
Chore(deps): Bump license.maven.plugin.version from 4.3 to 4.4

Bumps `license.maven.plugin.version` from 4.3 to 4.4.

Updates `com.mycila:license-maven-plugin-git` from 4.3 to 4.4
- [Commits](https://github.com/mathieucarbou/license-maven-plugin/compare/license-maven-plugin-4.3...license-maven-plugin-4.4)

Updates `com.mycila:license-maven-plugin` from 4.3 to 4.4
- [Commits](https://github.com/mathieucarbou/license-maven-plugin/compare/license-maven-plugin-4.3...license-maven-plugin-4.4)

---
updated-dependencies:
- dependency-name: com.mycila:license-maven-plugin-git
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: com.mycila:license-maven-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mwithi committed:
Merge branch 'develop' into JantBogard-OP-1193

mwithi reviewed (changes requested):
Please, I've aligned your fork/branch to the latest gui/develop, pull the changes in your local.

Note1: a new inventory on an emtpy DB doesn't seem to work
Note2: a new inventory on the demo DB it throws exception when trying to update the REAL QUANTITY field

```
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException: Cannot invoke "javax.swing.JTable.rowAtPoint(java.awt.Point)" because "this.this$0.table" is null
```

dbmalkovsky committed:
Merge pull request #1989 from informatici/dependabot/maven/license.maven.plugin.version-4.4

Chore(deps): Bump license.maven.plugin.version from 4.3 to 4.4

mwithi committed:
OP-1235 Reflecting changes in CORE (#1990)

* Fix with new constructors

Also fix naming test helper class conflicting with core

* Apply suggestions from code review

mwithi committed:
OP-1229 Medicals sorted by name in Pharmaceutical Stock (#1993)

mwithi committed:
Fix MedicalBrowser style and filtering (#1992)

- Fix CellRenderer using the actual showing data
- Fix MedicalBrowser filtering after edit
- Fix columns auto resizing in MedicalBrowser

mwithi committed:
OP-1228 Add product code column to Pharmaceutical Stock (#1994)

mwithi committed:
OP-1225 Allow to change MedicalType (#1995)

mwithi committed:
OP-1227 Show 2 decimals only for prices in Pharmaceutical Stock (#1996)

mwithi committed:
OP-1219 Add Ward comparator to sort in combobox (#1997)

mwithi committed:
OP-1220 Add Supplier comparator to sort in combobox (#1999)

mwithi committed:
OP-1221 Improve filtering for all charge or discharge MovementType(s) (#1998)

* OP-1221 Improve filtering for all charge or discharge MovementType(s)

* Apply suggestions from code review

* Factorize constant strings

---------

Co-authored-by: David B Malkovsky <david@malkovsky.org>

mwithi committed:
OP-1224 add reset button for pharmaceutical stock filter (#2000)

* OP-1224 Add Reset button for Pharmaceutical Stock filter

* Fix bundles

mwithi committed:
OP-1222 ProductLedger improvements (#2001)

mwithi committed:
OP-1223 Improve Pharmaceutical Stock label and graphics (#2002)

mwithi committed:
Update version

mwithi committed:
Fix pom

dependabot[bot] committed:
Chore(deps-dev): Bump org.mockito:mockito-junit-jupiter

Bumps [org.mockito:mockito-junit-jupiter](https://github.com/mockito/mockito) from 5.11.0 to 5.12.0.
- [Release notes](https://github.com/mockito/mockito/releases)
- [Commits](https://github.com/mockito/mockito/compare/v5.11.0...v5.12.0)

---
updated-dependencies:
- dependency-name: org.mockito:mockito-junit-jupiter
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Chore(deps): Bump license.maven.plugin.version from 4.4 to 4.5

Bumps `license.maven.plugin.version` from 4.4 to 4.5.

Updates `com.mycila:license-maven-plugin-git` from 4.4 to 4.5
- [Commits](https://github.com/mathieucarbou/license-maven-plugin/compare/license-maven-plugin-4.4...license-maven-plugin-4.5)

Updates `com.mycila:license-maven-plugin` from 4.4 to 4.5
- [Commits](https://github.com/mathieucarbou/license-maven-plugin/compare/license-maven-plugin-4.4...license-maven-plugin-4.5)

---
updated-dependencies:
- dependency-name: com.mycila:license-maven-plugin-git
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: com.mycila:license-maven-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

JantBogard commented:
> Please, I've aligned your fork/branch to the latest gui/develop, pull the changes in your local.
> 
> Note1: a new inventory on an emtpy DB doesn't seem to work Note2: a new inventory on the demo DB it throws exception when trying to update the REAL QUANTITY field
> 
> ```
> Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException: Cannot invoke "javax.swing.JTable.rowAtPoint(java.awt.Point)" because "this.this$0.table" is null
> ```

Ok I fix it

JantBogard committed:
Merge branch 'informatici:develop' into OP-1193

mwithi committed:
Update license headers

mwithi committed:
Remove -SNAPSHOT before release

dbmalkovsky committed:
Merge pull request #2004 from informatici/dependabot/maven/license.maven.plugin.version-4.5

Chore(deps): Bump license.maven.plugin.version from 4.4 to 4.5

dbmalkovsky committed:
Merge pull request #2003 from informatici/dependabot/maven/org.mockito-mockito-junit-jupiter-5.12.0

Chore(deps-dev): Bump org.mockito:mockito-junit-jupiter from 5.11.0 to 5.12.0

mwithi committed:
Merge branch 'develop' into release_1_14_2

mwithi committed:
Merge pull request #2006 from informatici/release_1_14_2

Merge release_1_4_2 (resolve conflicts)

dbmalkovsky committed:
OP-1239 re-enable logging to a physical file (#1991)

* OP-1239 re-enable logging to a physical file

* OP-1239 change `log4.properties` to `log4j2-spring.properties`

mwithi committed:
OP-1218 Add Medical soft deletion GUI management (#2009)

* Add Medical soft deletion GUI management

* Remove old way of tracking changes

* Remove old way of tracking changes

transifex-integration[bot] committed:
Updates for project OpenHospital (#2010)

* Translate bundle/language_en.properties in es

97% of minimum 95% translated source file: 'bundle/language_en.properties'
on 'es'.

Sync of partially translated files: 
untranslated content is included with an empty translation 
or source language content depending on file format

* Translate bundle/language_en.properties in sq

99% of minimum 95% translated source file: 'bundle/language_en.properties'
on 'sq'.

Sync of partially translated files: 
untranslated content is included with an empty translation 
or source language content depending on file format

* Translate bundle/language_en.properties in de

99% of minimum 95% translated source file: 'bundle/language_en.properties'
on 'de'.

Sync of partially translated files: 
untranslated content is included with an empty translation 
or source language content depending on file format

* Translate bundle/language_en.properties in it

99% of minimum 95% translated source file: 'bundle/language_en.properties'
on 'it'.

Sync of partially translated files: 
untranslated content is included with an empty translation 
or source language content depending on file format

* Translate bundle/language_en.properties in zh_CN

99% of minimum 95% translated source file: 'bundle/language_en.properties'
on 'zh_CN'.

Sync of partially translated files: 
untranslated content is included with an empty translation 
or source language content depending on file format

* Translate bundle/language_en.properties in ar

99% of minimum 95% translated source file: 'bundle/language_en.properties'
on 'ar'.

Sync of partially translated files: 
untranslated content is included with an empty translation 
or source language content depending on file format

* Translate bundle/language_en.properties in am_ET

97% of minimum 95% translated source file: 'bundle/language_en.properties'
on 'am_ET'.

Sync of partially translated files: 
untranslated content is included with an empty translation 
or source language content depending on file format

---------

Co-authored-by: transifex-integration[bot] <43880903+transifex-integration[bot]@users.noreply.github.com>

transifex-integration[bot] committed:
Update translations for project OpenHospital (#2012)

* Translate bundle/language_en.properties in sq

99% of minimum 95% translated source file: 'bundle/language_en.properties'
on 'sq'.

Sync of partially translated files: 
untranslated content is included with an empty translation 
or source language content depending on file format

* Translate bundle/language_en.properties in sq

99% of minimum 95% translated source file: 'bundle/language_en.properties'
on 'sq'.

Sync of partially translated files: 
untranslated content is included with an empty translation 
or source language content depending on file format

* Translate bundle/language_en.properties in sq

99% of minimum 95% translated source file: 'bundle/language_en.properties'
on 'sq'.

Sync of partially translated files: 
untranslated content is included with an empty translation 
or source language content depending on file format

* Translate bundle/language_en.properties in sq

99% of minimum 95% translated source file: 'bundle/language_en.properties'
on 'sq'.

Sync of partially translated files: 
untranslated content is included with an empty translation 
or source language content depending on file format

* Translate bundle/language_en.properties in sq

99% of minimum 95% translated source file: 'bundle/language_en.properties'
on 'sq'.

Sync of partially translated files: 
untranslated content is included with an empty translation 
or source language content depending on file format

* Translate bundle/language_en.properties in sq

100% translated source file: 'bundle/language_en.properties'
on 'sq'.

---------

Co-authored-by: transifex-integration[bot] <43880903+transifex-integration[bot]@users.noreply.github.com>

transifex-integration[bot] committed:
Update translations for project OpenHospital (#2014)

* Translate bundle/language_en.properties in de

99% of minimum 95% translated source file: 'bundle/language_en.properties'
on 'de'.

Sync of partially translated files: 
untranslated content is included with an empty translation 
or source language content depending on file format

* Translate bundle/language_en.properties in de

99% of minimum 95% translated source file: 'bundle/language_en.properties'
on 'de'.

Sync of partially translated files: 
untranslated content is included with an empty translation 
or source language content depending on file format

* Translate bundle/language_en.properties in de

99% of minimum 95% translated source file: 'bundle/language_en.properties'
on 'de'.

Sync of partially translated files: 
untranslated content is included with an empty translation 
or source language content depending on file format

* Translate bundle/language_en.properties in de

99% of minimum 95% translated source file: 'bundle/language_en.properties'
on 'de'.

Sync of partially translated files: 
untranslated content is included with an empty translation 
or source language content depending on file format

* Translate bundle/language_en.properties in de

99% of minimum 95% translated source file: 'bundle/language_en.properties'
on 'de'.

Sync of partially translated files: 
untranslated content is included with an empty translation 
or source language content depending on file format

* Translate bundle/language_en.properties in de

100% translated source file: 'bundle/language_en.properties'
on 'de'.

---------

Co-authored-by: transifex-integration[bot] <43880903+transifex-integration[bot]@users.noreply.github.com>

mwithi committed:
OP-1217 Add Medical Stock Feature (#2013)

* First commit

* Add new AMC_ver2 report

* Add Movement Type category field 'production' or 'inventory'

* Improve report clarity

* Add MedicalType category

* Fix javadoc

* Restore original report name

* Add hidden column for stock movement Excel export

* Refactor category admitted values

* Apply suggestions from code review

---------

Co-authored-by: David B Malkovsky <david@malkovsky.org>

mwithi committed:
Update Eclipse settings (#2018)

dbmalkovsky committed:
OP-1278 resource bundle addition (#2015)

dbmalkovsky committed:
Chore: update Intellij settings to match Eclipse (#2019)

dependabot[bot] committed:
Chore(deps): Bump org.apache.maven.plugins:maven-surefire-plugin

Bumps [org.apache.maven.plugins:maven-surefire-plugin](https://github.com/apache/maven-surefire) from 3.2.5 to 3.3.0.
- [Release notes](https://github.com/apache/maven-surefire/releases)
- [Commits](https://github.com/apache/maven-surefire/compare/surefire-3.2.5...surefire-3.3.0)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-surefire-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dbmalkovsky committed:
Hibernate renamed org.hibernate.type to org.hibernate.orm.jdbc.bind (#2021)

dependabot[bot] committed:
Chore(deps): Bump org.apache.maven.plugins:maven-jar-plugin

Bumps [org.apache.maven.plugins:maven-jar-plugin](https://github.com/apache/maven-jar-plugin) from 3.4.1 to 3.4.2.
- [Release notes](https://github.com/apache/maven-jar-plugin/releases)
- [Commits](https://github.com/apache/maven-jar-plugin/compare/maven-jar-plugin-3.4.1...maven-jar-plugin-3.4.2)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-jar-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dbmalkovsky committed:
Merge pull request #2022 from informatici/dependabot/maven/org.apache.maven.plugins-maven-jar-plugin-3.4.2

Chore(deps): Bump org.apache.maven.plugins:maven-jar-plugin from 3.4.1 to 3.4.2

dbmalkovsky committed:
Merge pull request #2020 from informatici/dependabot/maven/org.apache.maven.plugins-maven-surefire-plugin-3.3.0

Chore(deps): Bump org.apache.maven.plugins:maven-surefire-plugin from 3.2.5 to 3.3.0

dependabot[bot] committed:
Chore(deps-dev): Bump org.junit.jupiter:junit-jupiter-engine

Bumps [org.junit.jupiter:junit-jupiter-engine](https://github.com/junit-team/junit5) from 5.10.2 to 5.10.3.
- [Release notes](https://github.com/junit-team/junit5/releases)
- [Commits](https://github.com/junit-team/junit5/compare/r5.10.2...r5.10.3)

---
updated-dependencies:
- dependency-name: org.junit.jupiter:junit-jupiter-engine
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dbmalkovsky committed:
Merge pull request #2024 from informatici/dependabot/maven/org.junit.jupiter-junit-jupiter-engine-5.10.3

Chore(deps-dev): Bump org.junit.jupiter:junit-jupiter-engine from 5.10.2 to 5.10.3

dependabot[bot] committed:
Chore(deps-dev): Bump org.assertj:assertj-core from 3.25.3 to 3.26.0

Bumps [org.assertj:assertj-core](https://github.com/assertj/assertj) from 3.25.3 to 3.26.0.
- [Release notes](https://github.com/assertj/assertj/releases)
- [Commits](https://github.com/assertj/assertj/compare/assertj-build-3.25.3...assertj-build-3.26.0)

---
updated-dependencies:
- dependency-name: org.assertj:assertj-core
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Chore(deps): Bump org.apache.maven.plugins:maven-dependency-plugin

Bumps [org.apache.maven.plugins:maven-dependency-plugin](https://github.com/apache/maven-dependency-plugin) from 3.6.1 to 3.7.1.
- [Release notes](https://github.com/apache/maven-dependency-plugin/releases)
- [Commits](https://github.com/apache/maven-dependency-plugin/compare/maven-dependency-plugin-3.6.1...maven-dependency-plugin-3.7.1)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-dependency-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dbmalkovsky committed:
Merge pull request #2011 from informatici/dependabot/maven/org.assertj-assertj-core-3.26.0

Chore(deps-dev): Bump org.assertj:assertj-core from 3.25.3 to 3.26.0

dbmalkovsky committed:
Merge pull request #2023 from informatici/dependabot/maven/org.apache.maven.plugins-maven-dependency-plugin-3.7.1

Chore(deps): Bump org.apache.maven.plugins:maven-dependency-plugin from 3.6.1 to 3.7.1

dependabot[bot] committed:
Chore(deps-dev): Bump org.assertj:assertj-core from 3.26.0 to 3.26.3

Bumps [org.assertj:assertj-core](https://github.com/assertj/assertj) from 3.26.0 to 3.26.3.
- [Release notes](https://github.com/assertj/assertj/releases)
- [Commits](https://github.com/assertj/assertj/compare/assertj-build-3.26.0...assertj-build-3.26.3)

---
updated-dependencies:
- dependency-name: org.assertj:assertj-core
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dbmalkovsky committed:
Merge pull request #2025 from informatici/dependabot/maven/org.assertj-assertj-core-3.26.3

Chore(deps-dev): Bump org.assertj:assertj-core from 3.26.0 to 3.26.3

dependabot[bot] committed:
Chore(deps): Bump org.apache.maven.plugins:maven-surefire-plugin

Bumps [org.apache.maven.plugins:maven-surefire-plugin](https://github.com/apache/maven-surefire) from 3.3.0 to 3.3.1.
- [Release notes](https://github.com/apache/maven-surefire/releases)
- [Commits](https://github.com/apache/maven-surefire/compare/surefire-3.3.0...surefire-3.3.1)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-surefire-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dbmalkovsky committed:
Merge pull request #2026 from informatici/dependabot/maven/org.apache.maven.plugins-maven-surefire-plugin-3.3.1

Chore(deps): Bump org.apache.maven.plugins:maven-surefire-plugin from 3.3.0 to 3.3.1

mwithi committed:
OP-1301-fix-units-packets-management-in-main-store (#2027)

* Improve quantity check for single lot

* Add exception for AUTOMATICLOT_OUT case

* Apply project standards

* Deprecate initialQty

* Apply suggestions from code review

---------

Co-authored-by: David B Malkovsky <david@malkovsky.org>

JantBogard committed:
work on update inventory new specification

JantBogard committed:
work on update inventory new specification

JantBogard committed:
Merge commit '37499d719fa70875fc06209d57f38f3c35871377' into OP-1193

JantBogard committed:
Apply new specification in ward inventory

JantBogard committed:
chore:Remove unecessary MessageDialog

dbmalkovsky commented:
I can't tell what code you really changed; there is so much code that has been committed for weeks/months included in this PR.
Please start with a new branch that matches the current code base and then apply your changes.   That is the only way I can reasonably review what you are trying to accomplish.  Thanks

JantBogard commented:
> I can't tell what code you really changed; there is so much code that has been committed for weeks/months included in this PR. Please start with a new branch that matches the current code base and then apply your changes. That is the only way I can reasonably review what you are trying to accomplish. Thanks

Ok
